mstm/__init__.py

```diff
@@ -51,99 +51,109 @@ def separate_exponent(num):
+            angs.append(str(th[i])+ ',' +str(phi[i]))        
+        
+        angfile = open(os.path.join(temp_dir, AngfileName),'w')
+        for item in angs:
+            angfile.write('%s\n' %item)
+        angfile.close()
+        
+        OutputName = 'mstm_out.dat'
+        length_scl_factor = 2*np.pi/wl
+        polarization_angle = np.arctan2(inc.jones_vec[1],inc.jones_vec[0])
+        parameters = (targ.num_spheres, targ.index_spheres, targ.index_matrix, length_scl_factor, polarization_angle)
+        g = ''
+        for i in np.arange(0,targ.num_spheres):
+            radb, rade = separate_exponent(targ.radii[i]) #have to make sure we don't print any e's into the text file.
+            xb, xe = separate_exponent(targ.x[i]-np.mean(targ.x))
+            print xb
+            yb, ye = separate_exponent(targ.y[i]-np.mean(targ.y))
+            zb, ze = separate_exponent(-(targ.z[i]-np.mean(targ.z)))
+            g += '{0:20} {1:20} {2:20} {3:20}'.format(str(radb)+'d'+str(rade), str(xb)+'d'+str(xe), str(yb)+'d'+str(ye), str(zb)+'d'+str(ze))+'\n'
+        with open(templatelocation, 'r') as InFile:
+            InF = InFile.read()				
+        InF = InF.format(parameters, g, OutputName, AngfileName, len(angs) )
+        InputFile = file(os.path.join(temp_dir, 'mstm.inp'), 'w')
+        InputFile.write(InF)
+        InputFile.close()
+        
+        
+        # run MSTM
+        cmd = ['./mstm_ubuntu.exe', 'mstm.inp']
+        subprocess.check_call(cmd, cwd=temp_dir)
+        
+        # Go into results file
+        result_file = glob.glob(os.path.join(temp_dir,'mstm_out.dat'))[0]
+        print temp_dir
+        # Read correct lines of results file
+        mstm_result = iter(open(result_file))
+        S11 = []
+        S12 = []
+        S13 = []
+        S14 = []
+        S11col = -1
+        for line in mstm_result:
+            print line
             l = line.split()
-        if len(l)>0 and S11col>-1:
-            if 'matrix' in l:
-                break
-            if '************' in l[0]:
-                break
-            S11.append(float(l[S11col]))
-            S12.append(float(l[S11col + 1]))
-            S13.append(float(l[S11col + 2]))
-            S14.append(float(l[S11col + 3]))
-    Srow1 = [S11, S12, S13, S14]
-
-    os.chdir(current_directory)
-    if delete:
-        shutil.rmtree(temp_dir)
-
-    return Srow1, th, phi
+            if '11' in l:
+                S11col = l.index('11')
+                line = next(mstm_result)
+                l = line.split()
+            if len(l)>0 and S11col>-1:
+                if 'matrix' in l:
+                    break
+                if '************' in l[0]:
+                    break
+                S11.append(float(l[S11col]))
+                S12.append(float(l[S11col + 1]))
+                S13.append(float(l[S11col + 2]))
+                S14.append(float(l[S11col + 3]))
+        Srow1 = [S11, S12, S13, S14]
+        scat_mat_dict[wl] = [Srow1,th,phi]        
```

Comment:
not sure a dictionary is the right object to use here. You can use numpy arrays (np.array) for both the scattering matrix and the scattering matrices at each wavelength.  The advantage of using the arrays is that you can take advantage of [broadcasting](http://cs231n.github.io/python-numpy-tutorial/#numpy-broadcasting) to vectorize and speed up all computations.